### PR TITLE
eosio.contracts v1.6.1 upstream merge

### PR DESCRIPTION
**Pull request by ARoomWithABue**
## Change Description

* `eosio.tedp` -  A new contract designed to manage the funds in the `exrsrv.tf` account. This allows the producers to set timed contributions for the `tf`, `econdevfunds`, and `eosio.rex` funds.
* `eoxio.rex` - The eosio.contracts v1.6.1 rex changes have been completely merged and integrated.
* Unit Tests - Many of the eos unit tests needed slight refactoring due to the difference between eos and telos voting logic. This difference also affected the `eosio.rex` system unit tests. All tests are now passing, save one, which fails due to the setup the testing environment. @ARoomWithABue will be contacting b1 developers for guidance on this test.

## Deployment Changes
- [x] Deployment Changes

* The `exrsrv.tf` account must exist on the network and must have an `eosio.token::accounts` emplacement. Code in `producer_pay.cpp` in `eosio::onblock` checks the balance of `exrsrv.tf`. If the above requirements aren't met `eosio::onblock` could fail when calculating producer payments. 

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: contracts/eosio.amend/src/eosio.amend.cpp [hunk 1]
```diff
@@ -382,8 +385,6 @@ extern "C" {
             execute_action(name(self), name(code), &ratifyamend::cancelsub);
         } else if (code == self && action == name("addclause").value) {
             execute_action(name(self), name(code), &ratifyamend::addclause);
-        } else if (code == self && action == name("removeclause").value) {
```
Comment (kesar) [inline contracts/eosio.amend/src/eosio.amend.cpp]: this is removed but not the method ? why is removeclause removed?

File: contracts/eosio.trail/include/trail.voting.hpp [hunk 2]
```diff
@@ -15,6 +15,16 @@
 using namespace std;
 using namespace eosio;
 
+#pragma region Contants
```
Comment (kesar) [inline contracts/eosio.trail/include/trail.voting.hpp]: we shouldnt push commented code

File: contracts/eosio.amend/src/eosio.amend.cpp (hunk 1 above)
Comment (ARoomWithABue) [reply contracts/eosio.amend/src/eosio.amend.cpp]: The new `eosio.amend` changes weren't in the branch when @poplexity forked the branch. I've added the changes to the branch.

File: contracts/eosio.trail/include/trail.voting.hpp (hunk 2 above)
Comment (ARoomWithABue) [reply contracts/eosio.trail/include/trail.voting.hpp]: Agreed. I'll create another PR that will clean up the comments.

**Pull request merged**